File: src/Python/58.LengthOfLastWord.py
class Solution:
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        n = len(s)
        if n == 0:
            return 0
        i = n - 1
        length = 0
        while i >= 0 and s[i] == ' ':
            i -= 1
        while i >= 0:
            if s[i] != ' ':
                length += 1
            else:
                break
            i -= 1
        return length
        
if __name__ == '__main__':
    solution = Solution()
    print(solution.lengthOfLastWord('hello world'))
    print(solution.lengthOfLastWord('a'))
else:
    pass

File: src/Java/58.LengthOfLastWord.java
import java.awt.List;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;

class Solution {
    public int lengthOfLastWord(String s) {
        char[] arr = s.toCharArray();
        int n = arr.length;
        if (n == 0)
            return 0;
        int i = n - 1;
        int length = 0;
        while (i >= 0 && arr[i] == ' ') {
            --i;
        }
        while (i >= 0) {
            if (arr[i] != ' ')
                ++length;
            else
                break;
            --i;
        }
        return length;
    }
}
